insights/parsers/du_usage.py

```diff
@@ -0,0 +1,55 @@
+"""
+du - Command
+============
+
+Module for the processing of output from the ``du`` command.  The base class
+``DiskUsage`` provides all of the functionality for all classes.
+Data is avaliable as rows of the output contained
+in one ``Record`` object for each line of output.
+
+Sample input data for the ``du -sh`` command looks like::
+22G     /var/lib/mongodb
+
+"""
+
+from .. import Parser, parser, LegacyItemAccess
+
+
+def parse_du_lines(du_content):
+
+    du_output = {}
+    """Parse contents of each line in ``du`` output.
+    Parse each line of ``du`` output ensuring that wrapped lines are
+    reassembled prior to parsing, and that mount names containing spaces
+    are maintained.
+
+    Parameters:
+        du_content (list): Lines of du output to be parsed.
+
+    """
+    for line in du_content:
+        line_split = line.split()
+        du_output[line_split[1]] = line_split[0]
+
+    return du_output
+
+
+@parser("du_-h")
+class DiskUsage(LegacyItemAccess, Parser):
+    def __len__(self):
+        return len(self.data)
+
+    def parse_content(self, content):
+        self.data = parse_du_lines(content)
```

Comment:
Don't bother with having a separate function to do this, just put the function code here.  It needlessly obscures the flow of the code.

Reply:
done.

Reply:
I think @PaulWay must mean you'd better remove the `parse_du_lines` and shift its body into the `parse_content` directly.

Reply:
yup, I fixed it.

Reply:
Done.